### PR TITLE
Update changelog since v5.1.0

### PR DESCRIPTION
**Pull request by thomashoneyman**
**Description of the change**

This PR updates the CHANGELOG in preparation for the upcoming release of this library.

Related: purescript/purescript#3985

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,181 +5,204 @@ Notable changes to this project are documented in this file. The format is based
...
-- Non-empty combinators return `NonEmptyList` ([#102](https://github.com/purescript-contrib/purescript-parsing/pull/102))
+- Non-empty combinators now return `NonEmptyList` (#102)
+- Added support for PureScript 0.14 and dropped support for all previous versions (#103)
 
 New features:
 
 Bugfixes:
 
 Other improvements:
+- Changed default branch to `main` from `master`
```
Comment (JordanMartinez) [inline CHANGELOG.md:17]: Missing #98?

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,181 +5,204 @@ Notable changes to this project are documented in this file. The format is based
 ## [Unreleased]
 
 Breaking changes (😱!!!):
-- Non-empty combinators return `NonEmptyList` ([#102](https://github.com/purescript-contrib/purescript-parsing/pull/102))
+- Non-empty combinators now return `NonEmptyList` (#102)
+- Added support for PureScript 0.14 and dropped support for all previous versions (#103)
 
 New features:
```
Comment (JordanMartinez) [inline CHANGELOG.md:12]: Missing #87?

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,181 +5,204 @@ Notable changes to this project are documented in this file. The format is based
 ## [Unreleased]
 
 Breaking changes (😱!!!):
-- Non-empty combinators return `NonEmptyList` ([#102](https://github.com/purescript-contrib/purescript-parsing/pull/102))
+- Non-empty combinators now return `NonEmptyList` (#102)
+- Added support for PureScript 0.14 and dropped support for all previous versions (#103)
```
Comment (JordanMartinez) [inline CHANGELOG.md:9]: #101 did this. #103 might be a separate thing entirely.

Comment (JordanMartinez) [reply CHANGELOG.md:9]: Seems #93 is also missing.

Comment (JordanMartinez) [reply CHANGELOG.md:9]: Missing #97? Not sure if it's breaking.

**Review by JordanMartinez — CHANGES_REQUESTED** [review]
Seems to be a few PRs missing in this one.

File: CHANGELOG.md (hunk 3 above)
Comment (thomashoneyman) [reply CHANGELOG.md]: Good catches! I've added #101 and #93. #97 is included in the v5.1.0 release, so I've left it out. I've updated #103; I remember doing it as part of 0.14 changes but they're not specific to the 0.14 release, just a breaking change in a dependency.

File: CHANGELOG.md (hunk 2 above)
Comment (thomashoneyman) [reply CHANGELOG.md:14]: Added! Good catch.

File: CHANGELOG.md (hunk 1 above)
Comment (thomashoneyman) [reply CHANGELOG.md:20]: Added! Thanks.

**Pull request merged**